mixer/cmd/versions.go

```diff
@@ -0,0 +1,141 @@
+// Copyright © 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+
+	"github.com/clearlinux/mixer-tools/builder"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var updateVersionsCmd = &cobra.Command{
+	Use:   "update-versions",
+	Short: "Update the mix and upstream versions",
+	Long: `Choose the next mix and upstream versions based.
+
+Without any flags, it increments mix by 10 and upstream to the latest
+version of the same format used by current upstream. Use --mix-version
+and --upstream-version to set a specific version. By default the
+versions are shown but not set, to set use --set (-s).
```

Comment:
YMMV, but it feels more standard to me to make it set by default, and support a `--dry-run` flag for when people want to test.

Reply:
Yep. I even wrote a version with --dry-run but changed my mind with people "accidentaly" changing versions.

With the new scheme, I think that is less of an issue.